src/graph6.rs

```
@@ -0,0 +1,995 @@
+use crate::digraph6::{DiGraph, digraph_to_pydigraph};
+
+// End of combined module
+
+/// Convert internal Graph (undirected) to PyGraph
+fn graph_to_pygraph<'py>(py: Python<'py>, g: &Graph) -> PyResult<Bound<'py, PyAny>> {
+    let mut graph = StablePyGraph::<Undirected>::with_capacity(g.size(), 0);
+    // add nodes
+    for _ in 0..g.size() {
+        graph.add_node(py.None());
+    }
+    // add edges
+    for i in 0..g.size() {
+        for j in i..g.size() {
+            if g.bit_vec[i * g.size() + j] == 1 {
+                let u = NodeIndex::new(i);
+                let v = NodeIndex::new(j);
+                graph.add_edge(u, v, py.None());
+            }
+        }
+    }
+    let out = PyGraph {
+        graph,
+        node_removed: false,
+        multigraph: true,
+        attrs: py.None(),
+    };
+    Ok(out.into_pyobject(py)?.into_any())
+}
+
+// digraph_to_pydigraph provided by crate::digraph6
+
+/// Write a graph6 string to a file path. Supports gzip if the extension is `.gz`.
+pub(crate) fn to_file(path: impl AsRef<Path>, content: &str) -> std::io::Result<()> {
+    let extension = path
+        .as_ref()
+        .extension()
+        .and_then(|e| e.to_str())
+        .unwrap_or("");
+    if extension == "gz" {
+        let file = File::create(path)?;
+        let buf_writer = BufWriter::new(file);
+        let mut encoder = GzEncoder::new(buf_writer, Compression::default());
+        encoder.write_all(content.as_bytes())?;
+        encoder.finish()?;
+    } else {
+        std::fs::write(path, content)?;
+    }
+    Ok(())
+}
+
+#[pyfunction]
+#[pyo3(signature=(repr))]
+pub fn read_graph6_str<'py>(py: Python<'py>, repr: &str) -> PyResult<Bound<'py, PyAny>> {
+    // Wrap parser calls to catch Rust panics and convert IO errors to PyErr
+    // Helper enum used to return either a Graph or DiGraph from the parser.
+    enum ParserResult {
+        Graph(Graph),
+        DiGraph(DiGraph),
+    }
+
+    let wrapped = std::panic::catch_unwind(|| {
```

Do not use `std::panic::catch_unwind`. If you need to return an error use, `Result` as a return type with `Err`. Not `panic!` or `unwrap`. 

This will likely break with WASM.

---

Removed